lib/Query/QueryBuilderConverter.php

```diff
@@ -0,0 +1,256 @@
+                'e.g. $qb->from(\'a\') ...'
+            );
+        }
+
+        $this->applySourceConstraints($builder);
+
+        $phpcrQuery = $this->qomf->createQuery(
+            $this->from,
+            $this->constraint,
+            $this->orderings,
+            $this->columns
+        );
+
+        $query = new Query($phpcrQuery, $this->eventDispatcher, $this->locale, [], $builder->getPrimaryAlias());
+
+        if ($firstResult = $builder->getFirstResult()) {
+            $query->setFirstResult($firstResult);
+        }
+
+        if ($maxResults = $builder->getMaxResults()) {
+            $query->setMaxResults($maxResults);
+        }
+
+        return $query;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applySourceConstraints(QueryBuilder $builder)
+    {
+        foreach ($this->sourceDocumentNodes as $sourceDocumentNode) {
+            $metadata = $this->getMetadata($sourceDocumentNode->getDocumentFqn());
+            $constraint = $this->strategy->createSourceConstraint(
+                $this->qomf,
+                $sourceDocumentNode->getAlias(),
+                $metadata->getClass()
+            );
+
+            if (null === $constraint) {
+                continue;
+            }
+
+            if ($this->constraint) {
+                $this->constraint = $this->qomf->andConstraint(
+                    $this->constraint,
+                    $constraint
+                );
+                continue;
+            }
+
+            $this->constraint = $constraint;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function walkSourceDocument(SourceDocument $node)
+    {
+        $alias = $node->getAlias();
+        $documentFqn = $node->getDocumentFqn();
+
+        $this->sourceDocumentNodes[$alias] = $node;
+
+        $this->documentMetadata[$alias] = $this->getMetadata($documentFqn);
+
+        $alias = $this->qomf->selector(
+            $alias,
+            $this->strategy->getPrimaryNodeType($documentFqn)
+        );
+
+        return $alias;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function walkOperandDynamicField(OperandDynamicField $node)
+    {
+        $alias = $node->getAlias();
+        $field = $node->getField();
+
+        list($alias, $phpcrProperty) = $this->getPhpcrProperty(
+            $alias,
+            $field
+        );
+
+        $operand = $this->qomf->propertyValue(
+            $alias,
+            $phpcrProperty
+        );
+
+        return $operand;
+    }
+
+    /**
+     * Return the PHPCR property name and alias for the given document
+     * field name and query alias.
+     *
+     * @param string $alias As specified in the query source.
+     * @param string $field Name of the document field
+     *
+     * @return array {
+     *
+     *     @var string Element is the real alias to use, second element is
+     *     @var string the property name
+     * }
```

Comment:
I already do this elsewhere, if you want to change it it should be done globally.